### PR TITLE
PFM-4118-Time extension request changes in contract search and update.

### PR DESCRIPTION
**Pull request by shubhang-eGov**

### CODE REVIEW

File: backend/contracts/src/main/java/org/egov/works/enrichment/ContractEnrichment.java [hunk 1]
```diff
@@ -284,6 +330,17 @@ private void addLineItem(List<LineItems> refinedLineItems,LineItems lineItem,Lis
 
     private void enrichIdsAgreementDateAndAuditDetailsOnCreate(ContractRequest contractRequest) {
         Contract contract = contractRequest.getContract();
+        if (contract.getBusinessService() != null && contract.getBusinessService().equalsIgnoreCase(WORKFLOW_BUSINESS_SERVICE)) {
+            contract.setOldUuid(contract.getId());
```
Comment (shailesh-egov) [inline backend/contracts/src/main/java/org/egov/works/enrichment/ContractEnrichment.java:334]: we can not set oldUuid, versionNumber directly from request, first get the active contract from db and update the request

**Review by shailesh-egov — COMMENTED** [review]
@shubhang-eGov add logs

**Review by shailesh-egov — APPROVED** [review]

File: backend/contracts/src/main/java/org/egov/works/enrichment/ContractEnrichment.java [hunk 2]
```diff
@@ -317,6 +376,14 @@ private void enrichContractNumber(ContractRequest contractRequest) {
         contract.setContractNumber(generatedContractNumber);
         log.info("Contract Number enrichment is done. Generated Contract Number["+generatedContractNumber+"]");
     }
+    private void enrichSupplementNumber (ContractRequest contractRequest) {
+        String rootTenantId = contractRequest.getContract().getTenantId().split("\\.")[0];
```
[the remark this answers is not in the document]
Comment (shubhang-eGov) [reply backend/contracts/src/main/java/org/egov/works/enrichment/ContractEnrichment.java]: done

File: backend/contracts/src/main/java/org/egov/works/enrichment/ContractEnrichment.java (hunk 1 above)
Comment (shubhang-eGov) [reply backend/contracts/src/main/java/org/egov/works/enrichment/ContractEnrichment.java]: done

**Review by subhashini-egov — APPROVED** [review]

File: backend/contracts/src/main/java/org/egov/works/validator/ContractServiceValidator.java [hunk 3]
```diff
@@ -73,12 +78,21 @@ public void validateCreateContractRequest(ContractRequest contractRequest) {
...
-        validateCreateRequestedEstimateIdsAgainstEstimateServiceAndDB(contractRequest);
 
         // Validate orgId against Organization service data
         validateOrganizationIdAgainstOrgService(contractRequest);
 
+        if (contractRequest.getContract().getBusinessService() != null &&
+                contractRequest.getContract().getBusinessService().equalsIgnoreCase(WORKFLOW_BUSINESS_SERVICE)) {
+            log.info("Validating time extension request");
+            // Validate if Time Extension Request
+            validateTimeExtensionRequestForCreate(contractRequest);
```
[the remark this answers is not in the document]
Comment (shubhang-eGov) [reply backend/contracts/src/main/java/org/egov/works/validator/ContractServiceValidator.java]: done

**Pull request merged**